geomesa-dynamo/geomesa-dynamo-core/src/main/scala/org/locationtech/geomesa/dynamo/core/DynamoGeoQuery.scala

```diff
@@ -0,0 +1,126 @@
+    val (spatial, other) = partitionPrimarySpatials(query.getFilter, sft)
+    val iter: Iterator[SimpleFeature] =
+      if(query.equals(Query.ALL) || spatial.exists(FilterHelper.isFilterWholeWorld)) {
+        getAllFeatures(other)
+      } else {
+        val plans = planQuery(query, sft)
+        executeGeoTimeQuery(query, plans).toIterator
+      }
+    new DelegateSimpleFeatureReader(sft, new DelegateSimpleFeatureIterator(iter.asJava))
+  }
+
+  def getCountInternal(query: Query, sft: SimpleFeatureType): Int = {
+    if (query.equals(Query.ALL) || FilterHelper.isFilterWholeWorld(query.getFilter)) {
+      getCountOfAll
+    } else {
+      val plans = planQuery(query, sft)
+      executeGeoTimeCountQuery(query, plans).toInt
+    }
+  }
+
+  // Query Specific defs
+  def planQuery(query: Query, sft: SimpleFeatureType): GenTraversable[HashAndRangeQueryPlan] = {
+    val (lx, ly, ux, uy) = planQuerySpatialBounds(query)
+    val (dtgFilters, _) = partitionPrimaryTemporals(decomposeAnd(query.getFilter), sft)
+    val interval = FilterHelper.extractInterval(dtgFilters, sft.getDtgField)
+    val startWeeks: Int = DynamoPrimaryKey.epochWeeks(interval.getStart).getWeeks
+    val endWeeks:   Int = DynamoPrimaryKey.epochWeeks(interval.getEnd).getWeeks
+
+    val zRanges = DynamoPrimaryKey.SFC2D.toRanges(lx, ly, ux, uy).toList
+
+    val rows = (startWeeks to endWeeks).map { dt => getRowKeys(zRanges, interval, startWeeks, endWeeks, dt)}
+
+    val plans = rows.flatMap { case ((s, e), rowRanges) =>
+      rowRanges.map(row => planQueryForContiguousRowRange(s, e, row))
+    }
+    plans
+  }
+
+  def planQueryForContiguousRowRange(s: Int, e: Int, row: Int): HashAndRangeQueryPlan = {
+    val DynamoPrimaryKey.Key(_, _, _, _, z) = DynamoPrimaryKey.unapply(row)
+    val (minx, miny, maxx, maxy) = DynamoPrimaryKey.SFC2D.bound(z)
+    val min = DynamoPrimaryKey.SFC3D.index(minx, miny, s).z
+    val max = DynamoPrimaryKey.SFC3D.index(maxx, maxy, e).z
+    HashAndRangeQueryPlan(row, min, max, contained = false)
+  }
+
+  def getRowKeys(zRanges: Seq[IndexRange], interval: Interval, sew: Int, eew: Int, dt: Int): ((Int, Int), Seq[Int]) = {
+    val dtshift = dt << 16
+    val seconds: (Int, Int) =
+      if (dt != sew && dt != eew) {
+        (0, DynamoPrimaryKey.ONE_WEEK_IN_SECONDS)
+      } else {
+        val starts = if (dt == sew) DynamoPrimaryKey.secondsInCurrentWeek(interval.getStart) else 0
+        val ends   = if (dt == eew) DynamoPrimaryKey.secondsInCurrentWeek(interval.getEnd)   else DynamoPrimaryKey.ONE_WEEK_IN_SECONDS
+        (starts, ends)
+      }
+
+    val shiftedRanges = zRanges.flatMap { ir =>
+      val (l, u, _) = ir.tuple
+      (l to u).map { i => (dtshift + i).toInt }
+    }
+
+    (seconds, shiftedRanges)
+  }
+
+  def planQuerySpatialBounds(query: Query): (Double, Double, Double, Double) = {
+    val origBounds = query.getFilter.accept(ExtractBoundsFilterVisitor.BOUNDS_VISITOR, DefaultGeographicCRS.WGS84).asInstanceOf[Envelope]
```

Comment:
there should also be a follow on ticket that does a find a replace in path for this :smile: